File: Mentor.Me.Web/ClientApp/src/app/shared/types/proposition.ts
import PropositionType from './enums/proposition-type.enum';
import Skill from './skill';
import User from './user';

export default class Proposition {
    public id: string;
    public name: string;
    public description: string;
    public type: PropositionType;
    public startPrice: number;
    public active: boolean;
    public ownerId: string;
    public members: User[];
    public skills: Skill[];
}

File: Mentor.Me.Data/Entities/Proposition.cs
using Mentor.Me.Data.Enums;

namespace Mentor.Me.Data.Entities
{
    public class Proposition : IEntity
    {
        public Guid Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public PropositionType Type { get; set; }
        public decimal StartPrice { get; set; }
        public bool Active { get; set; }
        public Guid OwnerId { get; set; }
        public IEnumerable<User> Members { get; set; }
        public IEnumerable<Skill> Skills { get; set; }
    }
}